pymzml/spec.py

```diff
@@ -1282,13 +1285,13 @@ def remove_noise(self, mode="median", noise_level=None):
         # callcentPeaks = self.peaks("centroided")
         if noise_level is None:
             noise_level = self.estimated_noise_level(mode=mode)
-        if self._peak_dict["centroided"] is not None:
+        if len(self._peak_dict["centroided"]) != 0:
             self._peak_dict["centroided"] = self.peaks("centroided")[
-                self.peaks("centroided")[:, 1] >= noise_level
+                self.peaks("centroided")[:, 1]/noise_level >= signal_to_noise_threshold
             ]
-        if self._peak_dict["raw"] is not None:
+        if len(self._peak_dict["raw"]) != 0:
```

Comment:
As discussed, I think `len(self.peaks('centroided')) != 0` would be better

Reply:
Done